src/types/generic.ts

```diff
@@ -0,0 +1,262 @@
+import {HomeAssistant} from "custom-card-helpers";
+
+export namespace generic {
```

Comment:
I wonder if there is a risk overusing namespaces? 
Will prefixing all these types with `generic` make the code a little less concise and readable?
My experience says namespaces are used in huge codebases where identical names of types needs to be separated. I am not convinced namespaces are really needed in this codebase though.

Reply:
Just for now it's name-spaced to avoid such conflicts and for my own recognition.
It's a good comment though. The ones I need to optimize the code.

Reply:
Nope. I haven't gotten around to it yet and I really like to merge this PR since it disrupts other PRs.